src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -399,7 +399,10 @@ public bool KeyExists(byte[] key)
     public IBatch StartBatch()
     {
         IBatch batch = new RocksDbBatch(this);
-        _currentBatches.Add(batch);
+        lock (_currentBatches)
```

Comment:
maybe better to move it to Concurrent collection?

Reply:
I thought so too, but can't find a concurrent set.

Reply:
There is a very good nuget package ConcurrentHashSet, I used it once but then didn't need it anymore. Reviewed the code it was almost 1:1 to ConcurrentDictionary without overhead of values.

Reply:
Swtiched to it.